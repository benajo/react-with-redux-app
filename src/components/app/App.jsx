import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import HomePage from '../home/HomePage';
import AboutPage from '../about/About';
import Header from '../common/Header';
import PageNotFound from '../page-not-found/PageNotFound';
import { CoursesPage } from '../courses/CoursePage/CoursesPage';
import { ManageCoursePage } from '../courses/ManageCoursePage/ManageCoursePage';
import { AuthorsPage } from '../authors/AuthorPage/AuthorPage';
import { ManageAuthorPage } from '../authors/ManageAuthorPage/ManageAuthorPage';
import 'react-toastify/dist/ReactToastify.css';

const App = () => (
  <div className="container-fluid">
    <Header />
    <Switch>
      <Route exact path="/" component={HomePage} />
      <Route path="/about" component={AboutPage} />
      <Route path="/courses" component={CoursesPage} />
      <Route path="/course/:slug" component={ManageCoursePage} />
      <Route path="/course" component={ManageCoursePage} />
      <Route path="/authors" component={AuthorsPage} />
      <Route path="/author/:slug" component={ManageAuthorPage} />
      <Route path="/author" component={ManageAuthorPage} />
      <Route component={PageNotFound} />
    </Switch>
    <ToastContainer autoClose={3000} hideProgressBar />
  </div>
);

export default App;
