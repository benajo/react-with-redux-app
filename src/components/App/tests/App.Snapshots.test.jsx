import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';

it('render App component', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <App />
    </MemoryRouter>,
  );
  expect(tree).toMatchSnapshot();
});
