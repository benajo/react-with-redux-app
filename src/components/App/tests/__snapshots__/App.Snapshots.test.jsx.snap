// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App component 1`] = `
<div
  className="container-fluid"
>
  <nav>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "color": "#F15B2A",
        }
      }
    >
      Home
    </a>
     | 
    <a
      aria-current={null}
      href="/courses"
      onClick={[Function]}
    >
      Courses
    </a>
     | 
    <a
      aria-current={null}
      href="/authors"
      onClick={[Function]}
    >
      Authors
    </a>
     | 
    <a
      aria-current={null}
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </nav>
  <div
    className="jumbotron"
  >
    <h1>
      Pluralsight Administration
    </h1>
    <p>
      React, Redux and React Router for ultra-responsive web apps.
    </p>
    <a
      className="btn btn-primary btn-lg"
      href="/about"
      onClick={[Function]}
    >
      Learn more
    </a>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
